### ISSUE TITLE
WebApp:webapp access-restriction add with --service-tag throws None Type is not iterable error

### ISSUE THREAD

**Issue by Jay-Madden** [Azure/azure-cli#18500]
**Describe the bug**
Attempting to add a webapp access-restriction with the --service-tag argument throws `'NoneType' object is not iterable`
![image](https://user-images.githubusercontent.com/25359755/122119298-0508a180-cdf7-11eb-9885-f89de24127b6.png)

**To Reproduce**
Run the command from the [docs](https://docs.microsoft.com/en-us/cli/azure/webapp/config/access-restriction?view=azure-cli-latest), `az webapp config access-restriction add -g ResourceGroup -n AppName --priority 400 --service-tag AzureCloud`

**Expected behavior**
An access restriction is added with that service tag

**Environment summary**
```PS C:\Users\274086\source\repos\Environmental.EMIS\EMIS.IaC\Terraform> az --version
azure-cli                         2.25.0

core                              2.25.0
telemetry                          1.0.6

Extensions:
interactive                        0.4.4

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\274086\.azure\cliextensions'

Python (Windows) 3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 13:22:56) [MSC v.1928 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Additional context**
Attempting to add IP based access restrictions worked as expected

**Comment by yonzhan** [comment]
route to service team

**Comment by madsd** [comment]
@Jay-Madden would you mind running the command with --debug and post the result. If you prefer not to have the data displayed in the issue, you can send it to me directly: madsd[at]microsoft.com

**Comment by Jay-Madden** [comment]
@madsd Email sent

**Comment by madsd** [comment]
The issue occurs when caller does not have subscription permission to get list of Service Tags. I added a fix to skip validation if this is the case. #18527

**Comment by aparnavemuri** [comment]
Which version of the CLI will this fix be available? Will it be 2.26.0?
